fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/pdagg/PushDownAggregateCollector.java

```diff
@@ -0,0 +1,521 @@
+            CallOperator aggFn = entry.getValue();
+            ScalarOperator aggInput = aggFn.getChild(0);
+
+            if (!(aggInput instanceof CallOperator)) {
+                continue;
+            }
+
+            CallOperator callInput = (CallOperator) aggInput;
+            if (aggInput instanceof CaseWhenOperator) {
+                CaseWhenOperator caseWhen = (CaseWhenOperator) aggInput;
+                for (ScalarOperator condition : caseWhen.getAllConditionClause()) {
+                    condition.getUsedColumns().getStream().mapToObj(factory::getColumnRef)
+                            .forEach(v -> context.groupBys.put(v, v));
+                }
+
+                List<ScalarOperator> newWhenThen = Lists.newArrayList();
+                for (int i = 0; i < caseWhen.getWhenClauseSize(); i++) {
+                    if (caseWhen.getThenClause(i).isConstant() && !caseWhen.getThenClause(i).isConstantNull()) {
+                        // forbidden push down
+                        return visit(optExpression, context);
+                    }
+                    newWhenThen.add(ConstantOperator.createBoolean(false));
+                    newWhenThen.add(caseWhen.getThenClause(i));
+                }
+
+                // mock just value case when
+                CaseWhenOperator newCaseWhen = new CaseWhenOperator(caseWhen.getType(), null,
+                        caseWhen.hasElse() ? caseWhen.getElseClause() : null, newWhenThen);
+
+                // replace origin
+                aggFn.setChild(0, newCaseWhen);
+            } else if (callInput.getFunction() != null &&
+                    FunctionSet.IF.equals(callInput.getFunction().getFunctionName().getFunction())) {
+                if (aggInput.getChildren().stream().skip(1).anyMatch(c -> c.isConstant() && !c.isConstantNull())) {
+                    // forbidden push down
+                    return visit(optExpression, context);
+                }
+
+                aggInput.getChild(0).getUsedColumns().getStream().mapToObj(factory::getColumnRef)
+                        .forEach(v -> context.groupBys.put(v, v));
+                aggInput.setChild(0, ConstantOperator.createBoolean(false));
+            }
+        }
+
+        // check has constant aggregate, forbidden
+        if (!context.aggregations.isEmpty() &&
+                context.aggregations.values().stream().allMatch(ScalarOperator::isConstant)) {
+            return visit(optExpression, context);
+        }
+
+        return processChild(optExpression, context);
+    }
+
+    @Override
+    public Void visitLogicalAggregate(OptExpression optExpression, AggregateContext context) {
+        LogicalAggregationOperator aggregate = (LogicalAggregationOperator) optExpression.getOp();
+        // distinct/count* aggregate can't push down
+        if (aggregate.getAggregations().values().stream().anyMatch(c -> c.isDistinct() || c.isCountStar())) {
```

Comment:
what about `intersect_count` and `window_funnel` functions?

Reply:
will check in scan node